### PR TITLE
treewide: Remove Pelion DM support

### PR DESCRIPTION
**Pull request by pdunaj**
Remove integration with Pelion DM library.
Remove pelion_client application.

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

**Comment by pdunaj** [comment]
Hi @divipillai ,

It seems the reference being removed was used in old release notes. What should be done in such case? Should I simply swap the reference with a text?

**Review by emob-nordic — CHANGES_REQUESTED** [review]
Please have a look into `subsys/caf/modules/Kconfig.factory_reset_request` help.
```
-            boot-up. Factory reset will erase any stored Pelion and Network
-            data. It will also re-provision the device.

+            boot-up. Factory reset will erase any stored Network data.
+            It will also re-provision the device.
```

Also:
`subsys/caf/modules/net_state_lte.c:17`
`subsys/caf/modules/net_state_ot.c:20`
both use `CONFIG_PELION_CLIENT_NET_LOG_LEVEL`, but should probably use `CONFIG_CAF_INIT_LOG_NET_STATE_EVENTS`.

Should we also remove pelion references from known-isses?

**Comment by divipillai** [comment]
Yes, we just swap the reference with a text.

**Comment by pdunaj** [comment]
> Please have a look into `subsys/caf/modules/Kconfig.factory_reset_request` help.
> 
> ```
> -            boot-up. Factory reset will erase any stored Pelion and Network
> -            data. It will also re-provision the device.
> 
> +            boot-up. Factory reset will erase any stored Network data.
> +            It will also re-provision the device.
> ```
> 
> Also: `subsys/caf/modules/net_state_lte.c:17` `subsys/caf/modules/net_state_ot.c:20` both use `CONFIG_PELION_CLIENT_NET_LOG_LEVEL`, but should probably use `CONFIG_CAF_INIT_LOG_NET_STATE_EVENTS`.
> 
> Should we also remove pelion references from known-isses?

I have seen it and set up a ticket to fix that independently. I would like to keep this PR to minimum and focus on simply removing pelion - will be easier to review.

**Comment by emob-nordic** [comment]
> > Please have a look into `subsys/caf/modules/Kconfig.factory_reset_request` help.
> > ```
> > -            boot-up. Factory reset will erase any stored Pelion and Network
> > -            data. It will also re-provision the device.
> > 
> > +            boot-up. Factory reset will erase any stored Network data.
> > +            It will also re-provision the device.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Also: `subsys/caf/modules/net_state_lte.c:17` `subsys/caf/modules/net_state_ot.c:20` both use `CONFIG_PELION_CLIENT_NET_LOG_LEVEL`, but should probably use `CONFIG_CAF_INIT_LOG_NET_STATE_EVENTS`.
> > Should we also remove pelion references from known-isses?
> 
> I have seen it and set up a ticket to fix that independently. I would like to keep this PR to minimum and focus on simply removing pelion - will be easier to review.

Okay, let it be.

**Review by emob-nordic — APPROVED** [review]

**Comment by pdunaj** [comment]
Hi @divipillai , can we merge this one?

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -22,7 +22,7 @@ The most relevant changes that are present on the main branch of the |NCS|, as c
 Highlights
 **********
 
-* |no_changes_yet_note|
+* Removed support for Pelion DM library.
```
Comment (divipillai) [inline doc/nrf/releases/release-notes-changelog.rst:25]:
```suggestion
* Removed support for the Pelion DM library.
```

Comment (pdunaj) [reply doc/nrf/releases/release-notes-changelog.rst:25]: Quick question - can we fix it when release notes are moved? I am wondering if this justifies the CI rerun.

Comment (divipillai) [reply doc/nrf/releases/release-notes-changelog.rst:25]: Approving the PR, considering the Pelion will be removed from the known Issue in a separate ticket.

**Review by divipillai — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (pdunaj) [reply doc/nrf/releases/release-notes-changelog.rst:25]: I was rather thinking about leaving it as is (so we can use green CI) and add the missing `the` when there is a move from changelog to 1.8 release notes. There is always a review done at that moment where all such things are fixed. If you don't mind let's merge it.

Comment (SebastianBoe) [reply doc/nrf/releases/release-notes-changelog.rst:25]: merged

**Pull request merged**